components/feature/recentlyclosed/src/main/java/mozilla/components/feature/recentlyclosed/RecentlyClosedTab.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.recentlyclosed
+
+import android.content.Context
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.Engine
+
+/**
+ * A tab of a [TabCollection].
+ */
+interface RecentlyClosedTab {
+    /**
+     * Unique ID identifying this tab.
+     */
+    val id: Long
+
+    /**
+     * The title of the tab.
+     */
+    val title: String
+
+    /**
+     * The URL of the tab.
+     */
+    val url: String
+
+    /**
+     * Restores a single tab from this collection and returns a matching [Session].
+     *
+     * @param restoreSessionId If true the original [Session.id] of [Session]s will be restored. Otherwise a new ID
+     * will be generated. An app may prefer to use a new ID if it expects sessions to get restored multiple times -
+     * otherwise breaking the promise of a unique ID per [Session].
+     */
+    fun restoreRecentlyClosedTab(
```

Comment:
I mentioned this in the implementation comment, but leaving a note here for reference too: we should remove this since we shouldn't have the functionality of the tab try to restore itself, but rather, pass the serialized tab to the `BrowserStore` to be restored.

The `RecentlyClosedTab` should try to remain as close as possible to being like a data class.